lib/benchee/conversion/memory.ex

```diff
@@ -48,39 +48,48 @@ defmodule Benchee.Conversion.Memory do
 
   # Unit conversion functions
 
-  @type unit :: Unit.t
-  @type unit_atom :: atom
-  @type value_with_unit :: {number, unit}
+  @type unit_atom :: :byte | :kilobyte | :megabyte | :gigabyte | :terabyte
+  @type value_with_unit :: {number, Unit.t}
   @type value_with_unit_atom :: {number, unit_atom}
 
   @doc """
-  Converts a value for a specified unit and converts it to the equivalent of another unit of measure.
+  Converts a value for a specified unit atom and converts it to the equivalent of another unit of measure.
 
   ## Examples
 
-    iex> {value, unit} = Benchee.Conversion.Memory.convert({1024, :kilobyte}, :megabyte)
+    iex> {value, unit_atom} = Benchee.Conversion.Memory.convert({1024, :kilobyte}, :megabyte)
     iex> value
     1.0
-    iex> unit.name
+    iex> unit_atom
     :megabyte
   """
-  @spec convert(value_with_unit_atom, unit_atom) :: value_with_unit
+  @spec convert(value_with_unit_atom, unit_atom) :: unit_atom
   def convert({value, current_unit_atom}, desired_unit_atom) when is_atom(current_unit_atom) and is_atom(desired_unit_atom) do
     current_value_unit = {value, unit_for(current_unit_atom)}
     desired_unit = unit_for(desired_unit_atom)
-    convert(current_value_unit, desired_unit)
+    {new_value, _} = convert_with_unit(current_value_unit, desired_unit)
+    {new_value, desired_unit_atom}
   end
 
-  @spec convert(value_with_unit, unit) :: value_with_unit
-  def convert({value, %Unit{magnitude: current_magnitude}}, desired_unit = %Unit{magnitude: desired_magnitude}) do
+  @doc """
+  Converts a value for a specified %Unit and converts it to the equivalent of another unit of measure.
+
+  ## Examples
+
+    iex> kilobyte_unit = Benchee.Conversion.Memory.unit_for(:kilobyte)
+    iex> megabyte_unit = Benchee.Conversion.Memory.unit_for(:megabyte)
+    iex> {value, unit} = Benchee.Conversion.Memory.convert_with_unit({1024, kilobyte_unit}, megabyte_unit)
+    iex> value
+    1.0
+    iex> unit.name
+    :megabyte
+  """
+  @spec convert_with_unit(value_with_unit, Unit.t) :: value_with_unit
+  def convert_with_unit({value, %Unit{magnitude: current_magnitude}}, desired_unit = %Unit{magnitude: desired_magnitude}) do
```

Comment:
I'd just keep this as before as `convert/2` and an overloaded method that is pattern matching or what's the reason for changing it to be a different method? Imo that's better, we need to add some `|`s in the type spec but it should work :)